### ISSUE TITLE
Will check Dimenet++ implementation soon

### ISSUE THREAD

**Issue by thegodone** [aimat-lab/gcnn_keras#13]
I will try to check Dimenet++ implementation soon. I have made a PR on the PyG with fixed of the code but don't find time to refactor yet.

**Comment by thegodone** [comment]
code working but I will test with full dataset soon. I got this with your settings / datasize:

![image](https://user-images.githubusercontent.com/1186658/123555656-531e8d00-d787-11eb-9ef9-27317eb21870.png)

**Comment by PatReis** [comment]
DimeNet is not as fast as SchNet for example, I had about 170s per epoch. But the validation error is supposed to be much better for the full dataset. I had something like:
![dimnet_loss](https://user-images.githubusercontent.com/64005383/123610015-789bad00-d800-11eb-8e7e-113fee7b685e.png)

**Comment by thegodone** [comment]
I found a difference with original code they used "mad" instead of "std" for scaling the targets.

**Comment by PatReis** [comment]
Thanks, but I am not sure this makes a difference. I will continue to look into it. I found that the original dimenet++ also trains for the MAE as loss instead the MSE, which could yield better MAE error but also makes training a little bit more difficult. I changed to the MovingAverageAdam and an exponential learning rate scheduler. The original dimenet uses Gradient clipping, which I did not have used before and may not be easily accessible with standard keras.

**Comment by thegodone** [comment]
![image](https://user-images.githubusercontent.com/1186658/123781165-36dc3680-d8d4-11eb-8c95-4724f1276b82.png)

yes MAE is better for QM9 tasks. Gradient Clip is also important I go the case in my 2 days run using the full dataset too bad

**Comment by PatReis** [comment]
thanks for sharing. Let me do some testing. I will come back to you, if I have a good training script.

**Comment by PatReis** [comment]
Okay so I changed the training script and I got almost 20meV for LUMO on QM9. The things different are some initializations (which I already changed) and the embedding layer. I will also match the decay and epochs and run it again. The last thing different is the dataset. I think the source code of DimNet removed the unconverged molecules, which I haven't yet and use a certain split. If it dos not improve the next run, I will check this.
![dimnet_loss](https://user-images.githubusercontent.com/64005383/124439313-8f4e8080-dd79-11eb-9cb5-0042fb96e273.png)
